rust/src/database/verkle/keyed_update.rs

```diff
@@ -53,15 +57,90 @@ impl Ord for KeyedUpdate {
 
 impl KeyedUpdate {
     /// Returns the key associated with the update.
-    fn key(&self) -> &Key {
+    pub fn key(&self) -> &Key {
         match self {
             Self::FullSlot { key, .. } | Self::PartialSlot { key, .. } => key,
         }
     }
+
+    /// Applies the update to the given value.
+    #[cfg_attr(not(test), expect(unused))]
+    pub fn apply_to_value(&self, orig_value: &mut [u8; 32]) {
+        match self {
+            KeyedUpdate::FullSlot { value, .. } => {
+                *orig_value = *value;
+            }
+            KeyedUpdate::PartialSlot { value, mask, .. } => {
+                for i in 0..32 {
+                    orig_value[i] = (orig_value[i] & !mask[i]) | (value[i] & mask[i]);
+                }
+            }
+        }
+    }
 }
 
-impl From<Update<'_>> for Vec<KeyedUpdate> {
-    fn from(update: Update<'_>) -> Self {
+/// A collection of keyed updates with the invariants that they are sorted by key and non-empty.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct KeyedUpdates<'a>(Cow<'a, [KeyedUpdate]>);
```

Comment:
```suggestion
pub struct KeyedUpdateBatch<'a>(Cow<'a, [KeyedUpdate]>);
```

As discussed in person, maybe something like this to differentiate the two types a bit more.

Reply:
We could also consider moving away from the "update" terminology. This comes from FFI, but on the trie interface level I think it would make sense to e.g. `store(KeyValueBatch)`.